workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/context/NewAppContext.tsx

```diff
@@ -307,6 +307,26 @@ const reducer = (state: NewAppInitialState, action: AppActions) => {
 				tags?.includes(ProductTags.SOLUTION_PROFILE_APP_ICON)
 			);
 
+			const liferayPackages = Object.entries(
+				Object.groupBy(
+					_product.productVirtualSettings
+						.productVirtualSettingsFileEntries,
+					({version}) => version
+				)
+			).map((liferayPackage) => {
+				return {
+					files: liferayPackage[1]?.map((uploadedFile) => {
+						return {
+							error: false,
+							fileName: uploadedFile.src,
+							readableSize: '',
+							src: uploadedFile.src,
+						};
+					}),
+					version: liferayPackage[0],
+				};
+			});
+
```

Comment:
```suggestion
			const liferayPackages = Object.entries(
				Object.groupBy(
					_product.productVirtualSettings
						.productVirtualSettingsFileEntries,
					({version}) => version
				)
			).map(([version, files = []]) => ({
					files: files.map((uploadedFile) => ({
							error: false,
							fileName: uploadedFile.src,
							readableSize: '',
							src: uploadedFile.src,
						}),
					version
				});
```

Reply:
Pay attention to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy

As groupBy is `Newly available`